SearchTask/SearchTask.c

```diff
@@ -0,0 +1,323 @@
+﻿#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <stdbool.h>
+
+#define ARRAYLENGTH 10000
+
+void printArray(int array[], const int arrayLength)
+{
+    if (arrayLength == 0)
+    {
+        printf("None");
+        return;
+    }
+    for (int i = 0; i < arrayLength; i++)
```

Comment:
```suggestion
    for (int i = 0; i < arrayLength; ++i)
```
Тут и ниже. Это добрая сишная традиция, предпочитать префиксную форму